app/scripts/app-init.js

```diff
@@ -46,8 +50,26 @@ function importAllScripts() {
   );
 }
 
-// Placing script import call here ensures that scripts are inported each time service worker is activated.
-importAllScripts();
+// eslint-disable-next-line
+self.oninstall = () => {
```

Comment:
is there an alternative, like an event listener? who calls this?

Reply:
I took it from this example: https://stackoverflow.com/a/66408379/3696652

It is a service worker lifecycle callback, called once for MV3 service worker.

Reply:
As I am looking into fixing this issue, I found that we are on pretty old version of Web3.js `0.20.0`. Latest version of Web3.js already has this issue fixed. Should I just upgrade Web3 rather than patching it. It will be major version upgrade.

Reply:
This is called the first time the service worker is setup, but not after revived correct? Never understood why that was the case...

Reply:
prefer add listener is available

Reply:
Fixed now